storm-core/src/jvm/org/apache/storm/scheduler/utils/ArtifactoryConfigLoader.java

```diff
@@ -0,0 +1,429 @@
+    protected static final String ARTIFACTORY_BASE_DIRECTORY="artifactory.config.loader.base.directory";
+    protected static final String LOCAL_ARTIFACT_DIR="scheduler_artifacts";
+    static final String cacheFilename = "latest.yaml";
+
+    private static final Logger LOG = LoggerFactory.getLogger(ArtifactoryConfigLoader.class);
+
+    @SuppressWarnings("rawtypes")
+    private Map _conf;
+    private int _artifactoryPollTimeSecs = 600;
+    private boolean _cacheInitialized = false;
+    // Location of the file in the artifactory archive.  Also used to name file in cache.
+    private String _localCacheDir;
+    private String _artifactoryScheme = "http";
+    private String _baseDirectory = "/artifactory";
+    private int _lastReturnedTime = 0;
+    private int _timeoutSeconds = 10;
+    private Map _lastReturnedValue;
+
+    /**
+     * GETStringResponseHandler - A private class used to check the response
+     *                            coming back from httpclient.
+     *
+     */
+    private static class GETStringResponseHandler implements ResponseHandler<String> {
+        private static GETStringResponseHandler singleton = null;
+
+        /**
+         * @return a singleton httpclient GET response handler
+         */
+        public static GETStringResponseHandler getInstance() {
+            if (singleton == null) {
+                singleton = new GETStringResponseHandler();
+            }
+            return singleton;
+        }
+
+        @Override
+        /**
+         * @param response The http response to verify.
+         *
+         * @return null on failure or the response string if return code is in 200 range
+         */
+        public String handleResponse(final HttpResponse response) throws IOException {
+            int status = response.getStatusLine().getStatusCode();
+            if (status >= 200 && status < 300) {
+                HttpEntity entity = response.getEntity();
+                return entity != null ? EntityUtils.toString(entity) : null;
+            } else {
+                LOG.error("Got unexpected response code {}", status);
+                return null;
+            }
+        }
+    };
+
+    /**
+     * @param api null if we are trying to download artifact, otherwise a string to call REST api,
+     *        e.g. "/api/storage"
+     * @param artifact location of artifact
+     * @param host Artifactory hostname
+     * @param port Artifactory port
+     *
+     * @return null on failure or the response string if return code is in 200 range
+     * <p>
+     * Protected so we can override this in unit tests
+     */
+    protected String doGet(String api, String artifact, String host, Integer port) {
+        URIBuilder builder = new URIBuilder().setScheme(_artifactoryScheme).setHost(host).setPort(port);
+
+        String path = null;
+        if (api != null) {
+            path = _baseDirectory + "/" + api + "/" + artifact;
```

Comment:
there is no .path method in URIbuilder.  It's in javax.ws.rs.core.UriBuilder, which isn't pulled into storm anywhere.  I'd prefer to leave it as it, but will change and pull in new UriBuilder class if there is a consensus to do so.  @revans2?